### PR TITLE
Add Gateway6 to InterfaceStatus and NetworkInterfaceStatus

### PR DESCRIPTION
**Pull request by dhanvigoog**
Adds Gateway6 field to InterfaceStatus annotation struct and NetworkInterfaceStatus CRD type to support IPv6 gateways on secondary network interfaces in GKE Multi-Networking. Updated deepcopy functions and OpenAPI CRD manifests.

TESTED=Ran 'go test -v ./apis/network/v1/...', 'go vet ./...', and './hack/verify-codegen.sh' in gke-networking-api repo. TAG=agy

### CODE REVIEW

File: apis/network/v1/networkinterface_types.go [hunk 1]
```diff
@@ -72,6 +72,10 @@ type NetworkInterfaceStatus struct {
 	// Gateway4 defines the gateway IPv4 address for the network this interface connects to.
```
Comment (gsmanju1257) [inline apis/network/v1/networkinterface_types.go]: nit:  Let's add // +optional to Gateway4 as well since it will be unset on IPv6-only networks.

File: apis/network/v1/annotations_test.go [hunk 2]
```diff
@@ -339,6 +339,28 @@ func TestInterfaceStatusAnnotation(t *testing.T) {
 			},
 			expected: `[{"networkName":"network","ipAddresses":["1.2.3.4","ff::01"],"macAddress":"aa:bb:cc:dd:ee","routes":[{"to":"10.0.0.0/24"}],"gateway4":"10.0.0.1","dnsConfig":{"nameservers":["8.8.8.8"],"searches":["a.domain"]},"dhcpServerIP":"10.0.0.2"}]`,
 		},
+		{
```
Comment (gsmanju1257) [inline apis/network/v1/annotations_test.go:342]: in addition to the dual-stack test case, could you add an IPv6-only test case (where Gateway4 and DHCPServerIP are nil and only Gateway6 is set) just to verify gateway4 gets omitted when nil?

File: apis/network/v1/annotations_test.go [hunk 3]
```diff
@@ -339,6 +339,28 @@ func TestInterfaceStatusAnnotation(t *testing.T) {
...
+			name: "dual-stack status with gateway6",
+			input: InterfaceStatusAnnotation{
+				InterfaceStatus{
+					NetworkName: "network",
+					IPAddresses: []string{"1.2.3.4", "2001:db8::10"},
```
Comment (gsmanju1257) [inline apis/network/v1/annotations_test.go:347]: nit:  I guess we need to use CIDR format for IPAddresses ("1.2.3.4/32", "2001:db8::10/128") and a 6-byte MAC ("aa:bb:cc:dd:ee:ff") right for correct format?

File: apis/network/v1/networkinterface_types.go [hunk 4]
```diff
@@ -72,6 +72,10 @@ type NetworkInterfaceStatus struct {
...
 	Gateway4 *string `json:"gateway4,omitempty"`
 
+	// Gateway6 defines the gateway IPv6 address for the network this interface connects to.
```
Comment (gsmanju1257) [inline apis/network/v1/networkinterface_types.go]: nit: Let's add // +optional to Gateway4 as well since it will be unset on IPv6-only networks.

File: apis/network/v1/annotations_test.go (hunk 3 above)
Comment (dhanvigoog) [reply apis/network/v1/annotations_test.go]: Agree

One wrinkle worth flagging though: the format isn't actually uniform today. The
default-network branch in google_endpoint_api_manager.go writes bare IPs with no mask:

    if ipv4 := ep.GetIPv4Address(); ipv4 != "" {
        intfCR.Status.IpAddresses = append(intfCR.Status.IpAddresses, ipv4)
    }

So we have two producers disagreeing, and the field doc only says "IPAddresses are the IP
addresses assigned to this NIC. Can be either IPv4 or IPv6." I've matched the multi-nic
(CIDR) convention in the new tests since that's the path this Gateway6 change is for, but
I deliberately did not tighten the doc comment yet — declaring CIDR normative would make
that default-network writer retroactively non-conformant.

Do you want to (a) document IPAddresses as CIDR notation and treat the bare-IP path as a
bug to fix in anetd, or (b) leave the API permissive and have consumers tolerate both?

I'm thinking b as we don't want to reduce backwards compatibility by modifying the annotation.

Separately: the three pre-existing fixtures (minimal status / max status / multiple
interfaces) still use bare IPs and the 5-octet MAC. Leaving them here to keep this PR
scoped to Gateway6 — I'll clean them up in a follow-up CL.

File: apis/network/v1/annotations_test.go (hunk 2 above)
Comment (dhanvigoog) [reply apis/network/v1/annotations_test.go]: Good call — added as "ipv6-only status with gateway6".

It leaves Gateway4 and DHCPServerIP nil and asserts the marshalled JSON has neither key
at all (rather than "gateway4":null), with gateway6 landing in the slot gateway4 would
have occupied:

  [{"networkName":"network","ipAddresses":["2001:db8::10/128"],"macAddress":"aa:bb:cc:dd:ee:ff","routes":[{"to":"2001:db8::/64"}],"gateway6":"2001:db8::1","dnsConfig":{...}}]

Since the test round-trips through ParseInterfaceStatusAnnotation and cmp.Diffs back to
the input, this also covers unmarshalling gateway6 back into the *string.

File: apis/network/v1/annotations_test.go (hunk 3 above)
Comment (gsmanju1257) [reply apis/network/v1/annotations_test.go]: Okay that makes sense!

File: apis/network/v1/networkinterface_types.go (hunk 4 above)
Comment (gsmanju1257) [reply apis/network/v1/networkinterface_types.go]: Done.

**Review by gsmanju1257 — APPROVED** [review]
